internal/linter/check_pk_type.go

```diff
@@ -0,0 +1,69 @@
+package linter
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/skeema/skeema/internal/tengo"
+)
+
+func init() {
+	rule := Rule{
+		CheckerFunc:     TableBinaryChecker(pkTypeChecker),
+		Name:            "pk-type",
+		Description:     "Only allow primary keys to have types listed in --allow-pk-type",
+		DefaultSeverity: SeverityIgnore,
+	}
+	rule.RelatedListOption(
+		"allow-pk-type",
+		"tinyint, smallint, mediumint, int, bigint, varbinary, binary, date, datetime, timestamp, time, year, decimal",
+		"List of allowed data types for --lint-pk-type",
+		false, // permit empty list
+	)
+	RegisterRule(rule)
+}
+
+func pkTypeChecker(table *tengo.Table, createStatement string, _ *tengo.Schema, opts Options) *Note {
+	if table.PrimaryKey == nil {
+		// This checker is expected to be used in combination with check-pk
+		// so if there is no PK, we don't need to complain twice.
+		return nil
+	}
+	allowedTypes := opts.AllowList("pk-type")
+	if len(allowedTypes) == 0 {
+		return nil // disable check if allow-pk-type is empty
+	}
+	allowedStr := strings.Join(allowedTypes, ", ")
+	cols := table.ColumnsByName()
+	for _, part := range table.PrimaryKey.Parts {
+		if col, ok := cols[part.ColumnName]; ok {
+			colType := baseColType(col.TypeInDB)
+			if !opts.IsAllowed("pk-type", colType) {
+				message := fmt.Sprintf(
+					"Column %s of table %s is using data type %s, which is not configured to be permitted. The following data types are listed in option allow-pk-type: %s.",
```

Comment:
I'd recommend a slight edit for clarity, just to emphasize the problem is that the column type shouldn't be present specifically in a PK: "... not configured to be permitted in a primary key. The following data types ..."

Granted, this would be clear already from the Note.Summary, but Skeema CLI currently doesn't output that field anywhere. Once JSON output is supported in the future, it will be present there though.